src/components/Channel/Channel.js

```diff
@@ -138,6 +138,7 @@ const ChannelInner = ({
   const lastRead = useRef(new Date());
   const chatContext = useContext(ChatContext);
   const online = useRef(true);
+  const isMounted = useRef(true);
```

Comment:
I have seen this weird issue with our previous components as well. LGTM

@ath92 Maybe can we extract it to separate hook? 

```js
const isMounted = useIsMounted();
```

Reply:
I suppose this hook would basically be a combination of `useRef` for storing the flag + `useEffect` for setting it to true/false?

Reply:
yeah

Reply:
Just did this, also allowed me to write a test quite easily :)